modules/ROOT/pages/batch-phases.adoc

```diff
@@ -0,0 +1,135 @@
+. After processing records in a given array, the Batch Aggregator component sends all the records back to the stepping queue.
+
+* Behavior when the Batch Aggregator component is configured to stream records:
++
+In this scenario, the Batch Job component sends the processed records to its Batch Aggregator component. The Batch Aggregator component loads an array with  records from the stepping queue until all records from the current Batch Step for a given batch job instance are aggregated.
++
+A processor within the component must be able to accept an array of records as input.
++
+image:mruntime-batch-job-process-aggregator-streaming.png[Batch job process with Aggregator configured for Streaming]
++
+[calloutlist]
+. Each Batch Step components receives one or more record blocks and starts processing them in parallel.
+. After the Batch Step processes a record, the Batch Step sends the record to Batch Aggregator for further processing.
+. The Batch Aggregator component continues processing records until there are no more records in the stepping queue that are waiting to be processed by the current Batch Step.
+. The Batch Aggregator component sends all the aggregated records to the stepping queue.
+
+Mule retains a list of all records that succeed or fail to process through each Batch Step. If an event processor in a Batch Step fails to process a record, Mule continues processing the batch, skipping over the failed record in each subsequent Batch Step. The Batch Job component provides the `maxFailedRecords` property for setting the number records that can fail before the batch job stops. See xref:batch-reference.adoc#ref_batch_job_general_fields[Batch Job Properties].
+
+At the end of this phase, the batch job instance completes and ceases to exist.
+
+[[phase_on_complete]]
+== On Complete Phase
+
+During this phase, you can configure the runtime to create a report or summary of the records it processed in a given batch job instance. This phase provides system administrators and developers insight into which records failed or succeeded, but it does not process or provide access to individual records, nor does it pass processed records to downstream processors in the flow.
+
+As a best practice, configure a mechanism for reporting on failed or successful records to facilitate further action where required. During the On Complete phase, you can perform either of these tasks:
+
+* Reference the result object for the batch job instance from elsewhere in the Mule application to capture and use batch metadata, such as the number of records that failed to process in a particular batch job instance.
```

Comment:
TODO: NEED TO SHOW HOW TO DO THIS. Are we storing in object store or something from within <on-complete /> and calling it later?